cmd/dra-example-controller/gpu.go

```diff
@@ -110,15 +110,13 @@ func (g *gpudriver) allocate(crd *nascrd.NodeAllocationState, pod *corev1.Pod, g
 	available := make(map[string]*nascrd.AllocatableGpu)
 
 	for _, device := range crd.Spec.AllocatableDevices {
-		switch device.Type() {
-		case nascrd.GpuDeviceType:
+		if device.Type() == nascrd.GpuDeviceType {
```

Comment:
Let's keep the switch.

Reply:
What would the switch switch? We cannot even return an error from default in here, we just trust that there's only one type; so the default action will have to be empty like above, would that be fine ?

Reply:
Then change `allocate` so that it can return an `error`. Assuming that "allocation never fails" already seems suspicious.

Reply:
If you feel that a blank default is the right solution, then at least add a comment explaining to the reader/reviewer why the case is empty.

Reply:
That's exactly what I did. Should have put it long ago, you're right.